enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1273 @@
+
+`change_management_last_change`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+
+Value: 
+- `-1`: Datetime unknown. 
+- `0`: Material changes are currently permitted.
+- `> 0`: The number of seconds which have elapsed since the material changes were last permitted.
+
+`change_management_strategy_enabled`
+Labels:
+- kind=ClusterVersion|MachineConfigPool|HostedCluster|NodePool
+- object=<object-name>
+- system=<control-plane|worker-nodes>
+- strategy=MaintenanceSchedule|Manual|Assisted
+
+Value: 
+- `0`: Change management for this resource is not subject to this enabled strategy (**does** consider hierarchy based disable).
+- `1`: Change management for this resource is directly subject to this enabled strategy.
+- `2`: Change management for this resource is indirectly subject to this enabled strategy (i.e. only via control-plane override hierarchy).
+- `3`: Change management for this resource is directly and indirectly subject to this enabled strategy.
+
+#### Change Management Status
+Each resource which exposes a `.spec.changeManagement` stanza must also expose `.status.changeManagement` . 
+
+```yaml
+status:
+  changeManagement:
+    # Always show control-plane level strategy. Disabled if disabledUntil is "true".
+    clusterStrategy: <Disabled|MaintenanceSchedule>
+    # If this a worker-node related resource (e.g. MCP), show local strategy. Disabled if disabledUntil is "true".
+    workerNodeStrategy: <Disabled|MaintenanceSchedule|Manual|Assisted>
+    # Show effective state.
+    effectiveState: <Changes Paused|Changes Permitted>
+    description: "Human readable message explaining how strategies & configuration are resulting in the effective state."
+    # The start of the next permissive window, taking into account the hierarchy. "N/A" for indefinite pause.
+    permitChangesETA: <datetime>
+    changesPending: <Yes|No>
+```
+
+#### Change Management Bypass Annotation
+In some situations, it may be necessary for a MachineConfig to be applied regardless of the active change
+management policy for a MachineConfigPool. In such cases, `machineconfiguration.openshift.io/bypass-change-management`
+can be set to any non-empty string. The MCO will progress until MCPs which select annotated
+MachineConfigs have all machines running with a desiredConfig containing that MachineConfig's current state.
+
+This annotation will be present on `00-master` to ensure that, once the CVO updates the MachineConfig,
+the remainder of the control-plane update will be treated as a single material change.
+
+### Special Handling
+
+#### Change Management on Master MachineConfigPool
+In order to allow control-plane updates as a single material change, the MCO will only honor change the management configuration for the 
+master MachineConfigPool if user generated MachineConfigs are the cause for a pending change. To accomplish this, 

```

Comment:
We don't really have a mechanism for this today, since all configuration change calculations are done via rendered ones, and we just do a merge of all base configs.

I'm not sure I fully understand this section. Basically any pool-level configurations on the master pool are allowed the same way as worker, but they aren't honored if there's an incoming upgrade? But any user changes turn us back into worker calculation? (where the CVO level maint. schedule and the master pool are union'ed?)